dataland-backend/src/main/kotlin/org/dataland/datalandbackend/annotations/AnnotationProcessor.kt

```diff
@@ -0,0 +1,24 @@
+package org.dataland.datalandbackend.annotations
+
+import org.reflections.Reflections
+
+/**
+ * Class to process custom annotations
+ */
+class AnnotationProcessor {
+
+    /**
+     * Method to list all the permissible data types
+     * @return list of all permissible data types
+     */
+    fun getAllDataTypes(): List<String> {
+        val dataTypes: MutableList<String> = mutableListOf()
+        val reflections = Reflections("org.dataland.datalandbackend.model")
+        val allClassesWithDataTypeAnnotation = reflections.getTypesAnnotatedWith(DataTypeAnnotation::class.java)
+        for (clazz in allClassesWithDataTypeAnnotation) {
+            val datatype = clazz.simpleName
```

Comment:
Wir sollten konsistent bleiben - in DataController haben wir (für das gleiche): `    private val dataType = clazz.toString().substringAfterLast(".")`

Bitte an allen stellen die gleiche Methode benutzen.

Reply:
wurde in DataController mit `private val dataType = clazz.simpleName` ersetzt

Reply:
ok